src/rez/cli/pip.py

```diff
@@ -29,17 +38,129 @@ def setup_parser(parser, completions=False):
 
 def command(opts, parser, extra_arg_groups=None):
     from rez.pip import pip_install_package, run_pip_command
+    from rez.vendor.version.version import Version, VersionRange
     import sys
+    import json
+    import urllib2
+    import itertools
 
-    if not (opts.search or opts.install):
-        parser.error("Expected one of: --install, --search")
+    if not (opts.search or opts.install or opts.list or opts.query or opts.constrain):
+        parser.error("Expected one of: --install, --search, --list, --query, --constrain")
 
     if opts.search:
         p = run_pip_command(["search", opts.PACKAGE])
         p.wait()
         return
 
-    installed_variants, skipped_variants = pip_install_package(
+    if opts.list:
+        opts.PACKAGE += "=="
+        output = run_pip_command(["install", opts.PACKAGE], process_output=True)
+        for package_version in output:
+            print(package_version)
+        return
+
+    if opts.query:
+        URL = "https://pypi.python.org/pypi/{0}/{1}/json"
```

Comment:
Hello Jc good point. I was thinking that anyone using rez-pip would have an active internet connection but did not consider internal PyPI indexes. Although the query option help clearly states that is for use with PyPi.

If we add the URL as an option how will the query work? I am assuming we will need to add implementations for each method? A bit unfamiliar with internal index since I never used one.